### PR TITLE
[ELF] Rename target-specific RelExpr enumerators

### PR DESCRIPTION
**Pull request by MaskRay**
RelExpr enumerators are named `R_*`, which can be confused with ELF
relocation type names. Rename the target-specific ones to `RE_*` to
avoid confusion.

For consistency, the target-independent ones can be renamed as well, but
that's not urgent. The relocation processing mechanism with RelExpr has
non-trivial overhead compared with mold's approach, and we might make
more code into Arch/*.cpp files and decrease the enumerators.

### CODE REVIEW

**Comment by jrtc27** [comment]
Perhaps RE_* to match RelExpr? REL_ feels a lot like actual ELF things still.

**Comment by MaskRay** [comment]
> Perhaps RE_* to match RelExpr? REL_ feels a lot like actual ELF things still.

Thanks for the suggestion. `RE_*` looks better. Updated.

**Review by smithp35 — APPROVED** [review]
LGTM on the Arm and AArch64 side. I agree that a different prefix to R_ is preferred and RE_ is good enough for that.

File: lld/ELF/Relocations.h [hunk 1]
```diff
@@ -90,41 +90,41 @@ enum RelExpr {
   // Even though RelExpr is intended to be a target-neutral representation
   // of a relocation type, there are some relocations whose semantics are
   // unique to a target. Such relocation are marked with R_<TARGET_NAME>.
```
[the remark this answers is not in the document]
Comment (MaskRay) [reply lld/ELF/Relocations.h]: Thx. Updated

**Pull request merged**